tests/MiddlewareRunnerTest.php

```diff
@@ -90,6 +90,50 @@ public function testProcessStack(array $middlewares, $expectedCallCount)
         }
     }
 
+    public function provideThrownForTestCorruptMiddlewareStack()
+    {
+        return array(
+            array(0),
+            array(1),
+        );
+    }
+
+    /**
+     * @dataProvider provideThrownForTestCorruptMiddlewareStack
+     */
+    public function testCorruptMiddlewareStack($thrownCount)
+    {
+        $retry = function ($request, $next) {
+            return $next($request)->then(null, function ($error) use ($request, $next) {
+                // the $next failed. discard $error and retry once again:
+                return $next($request);
+            });
+        };
+
+        $runner = new MiddlewareRunner(array(
+            $retry,
+            function ($request, $next) use (&$thrownCount) {
+                if ($thrownCount <= 1) {
+                    $thrownCount++;
+                    throw new \RuntimeException('bingo');
+                }
```

Comment:
It looks like this could use a test :-D Does it make sense to you to split this into two dedicated tests and remove this logic? :+1:

Reply:
And I forgot what exactly this was supposed to test >_<!

Reply:
IIRC it was to ensure `$next` can be called more then once in case of retries.

Reply:
Got it, updated the PR :tada: